mage_ai/orchestration/db/__init__.py

```diff
@@ -51,10 +52,25 @@
     db_kwargs['pool_size'] = 50
     db_kwargs['connect_args']['options'] = '-c timezone=utc'
 
-engine = create_engine(
-    db_connection_url,
-    **db_kwargs,
-)
+try:
+    engine = create_engine(
+        db_connection_url,
+        **db_kwargs,
+    )
+    engine.connect()
+except SQLAlchemyError:
+    engine.dispose()
+    url_parsed = urlparse(db_connection_url)
+    if url_parsed.password:
+        db_connection_url = db_connection_url.replace(
+            url_parsed.password,
+            quote_plus(url_parsed.password),
+        )
+    engine = create_engine(
```

Comment:
If there's no url_parsed.password, we can directly throw exception, right?

Reply:
oops i didn't see this, i'll add it as a follow up